lib/graphql/language/parser.rb

```diff
@@ -608,7 +608,7 @@ def parse_enum_name
       end
 
       def parse_type_name
-        TypeName.new(pos: pos, name: parse_name, filename: @filename, source_string: @graphql_str)
+        TypeName.new(pos: pos, name: -parse_name, filename: @filename, source_string: @graphql_str)
```

Comment:
I see your trace as `graphql/c_parser` in it, which has its own parser and lexer. So I doubt this would affect the outcome of it. My guess would be here, where the C lexer makes tokens out of identifiers in the string: 

https://github.com/rmosolgo/graphql-ruby/blob/ec1ae00fdd36d5292096e40adb7401c4832bb7ee/graphql-c_parser/ext/graphql_c_parser_ext/lexer.rl#L264

https://github.com/rmosolgo/graphql-ruby/blob/ec1ae00fdd36d5292096e40adb7401c4832bb7ee/graphql-c_parser/ext/graphql_c_parser_ext/lexer.rl#L205-L209

Specifically the  `rb_utf8_str_new(...)` call. Maybe it would benefit from a `str_uminus` ([src](https://github.com/ruby/ruby/blob/4960a598d6bc42adcfd488c7f455b2defcdc8161/string.c#L3084)) call too?

Reply:
I see that this one is also a "repeat offender" in your profile: 

https://github.com/rmosolgo/graphql-ruby/blob/ec1ae00fdd36d5292096e40adb7401c4832bb7ee/lib/graphql/language/lexer.rb#L80